pkg/source/openstack/client.go

```diff
@@ -528,11 +547,24 @@ func (c *Client) checkOrGetUUID(input string) (string, error) {
 		return "", fmt.Errorf("error extracting servers in checkorgetuuid:%v", err)
 	}
 
-	if len(allServers) > 1 {
-		return "", fmt.Errorf(NotUniqueName)
+	if len(allServers) == 0 {
+		return allServers[0].ID, nil
+	}
```

Comment:
It'll have index out of range panic.

Reply:
this check is not needed since the pg length check checks if returned page is already empty so will drop it
```pg := servers.List(c.computeClient, servers.ListOpts{Name: input})
	allPg, err := pg.AllPages()
	if err != nil {
		return "", fmt.Errorf("error generating all pages in checkorgetuuid :%v", err)
	}
ok, err := allPg.IsEmpty()
	if err != nil {
		return "", fmt.Errorf("error checking if pages were empty in checkorgetuuid: %v", err)
	}

	if ok {
		return "", fmt.Errorf(NotServerFound)
	}
```